### PR TITLE
feat: override repository's visibility

### LINKED ISSUE #699: Add a field to overload repositories privacy
1. Add `forcePrivate` field in repository table.
2. Add a modifier in Repository model that use `forcePrivate || private`
3. Add a checkbox to force private privacy in repository settings.

### PR DESCRIPTION
**Pull request by jsfez**
fix #699

### CODE REVIEW

**Review by gregberge — CHANGES_REQUESTED** [review]
Are you sure that it is OK for in consumption usage?

**Comment by jsfez** [comment]
You were right, I forgot to count the consumption of forced private repositories screenshots as paid. It's fixed now

File: apps/database/src/models/Repository.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export class Repository extends Model {
       organizationId: { type: ["string", "null"] },
       userId: { type: ["string", "null"] },
       private: { type: "boolean" },
+      forcedPrivate: { type: ["boolean"] },
```
Comment (gregberge) [inline apps/database/src/models/Repository.ts:31]:
```suggestion
      forcedPrivate: { type: "boolean" },
```

File: apps/database/migrations/20230102064502_add_forced_private.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @param {import('knex').Knex} knex
+ */
+export const up = async (knex) => {
+  await knex.schema.table("repositories", (table) => {
+    table.boolean("forcedPrivate").defaultTo("false");
```
Comment (gregberge) [inline apps/database/migrations/20230102064502_add_forced_private.js:6]:
```suggestion
    table.boolean("forcedPrivate").notNullable().defaultTo("false");
```

File: apps/database/src/models/Repository.ts [hunk 3]
```diff
@@ -148,7 +150,7 @@ export class Repository extends Model {
   }
 
   static async checkReadPermission(repository: Repository, user: User | null) {
-    if (!repository.private) return true;
+    if (!repository.private || !repository.forcedPrivate) return true;
```
Comment (gregberge) [inline apps/database/src/models/Repository.ts:153]:
```suggestion
    if (!repository.private && !repository.forcedPrivate) return true;
```

File: apps/graphql/src/definitions/Repository.ts [hunk 4]
```diff
@@ -203,5 +210,21 @@ export const resolvers = {
...
+      return transaction(async (trx) => {
+        await checkUserRepositoryAccess({
+          user: ctx.user,
+          repositoryId: args.repositoryId,
+          trx,
+        });
+        return Repository.query(trx).patchAndFetchById(args.repositoryId, {
+          forcedPrivate: args.forcedPrivate,
+        });
+      });
```
Comment (gregberge) [inline apps/graphql/src/definitions/Repository.ts:227]: Transaction is useless.

**Review by gregberge — CHANGES_REQUESTED** [review]
- In `OwnerSettings.tsx` the private split have to be updated
- In `ReviewButton.tsx` the private check to disable the button have to be updated

**Review by gregberge — APPROVED** [review]

**Pull request merged**